autograd/nn.py

```diff
@@ -168,25 +181,38 @@ def forward(self, x):
         windows, (H_out, W_out) = extract_windows(
             x, self.kernel_size, self.stride, self.padding_mode
         )
-        # now windows: (b, c, H_out, W_out, kH, kW)
+        # windows: (H_out, W_out, N, C, H, W)
+
+        # Permute windows to (N, C, H, W, H_out, W_out)
+        windows = windows.permute(2, 3, 4, 5, 0, 1)
```

Comment:
This line was the crux of the "not learning problem". Other changes in this forward function is to ensure we maintain the same output shape (N, channel, H, W) as Pytorch, when we run unit tests to compare results.